src/plays/shopping-cart/reducer.tsx

```diff
@@ -0,0 +1,66 @@
+import { CartItem, InitialState } from "ShoppingCartTypes";

```

Comment:
Since reducer is a pure typescript file without having any react code in it, the file extension should be `.ts` (instead of `.tsx`)